pyuoi/decomposition/CUR.py

```diff
@@ -0,0 +1,158 @@
+from sklearn.base import BaseEstimator
+from sklearn.decomposition import TruncatedSVD
+from sklearn.model_selection import train_test_split
+
+import numpy as np
+
+
+class CUR(BaseEstimator):
+    def __init__(
+        self, n_resamples, max_k, resample_frac, algorithm='randomized',
+        n_iter=5, tol=0.0, random_state=None
+    ):
+        """Performs column subset selection (CUR decomposition) in the Union
+        of Intersections framework.
+
+        See Bouchard et al., NIPS, 2017, for more details on the Union of
+        Intersections framework.
+
+        Parameters
+        ----------
+        n_resamples : int
+            Number of resamples (bootstraps).
+
+        max_k : int
+            The maximum rank of the singular value decomposition.
+
+        resample_frac : float
+            The fraction of data to use in the resample.
+
+        algorithm : string, default = “randomized”
+            SVD solver to use. Either “arpack” for the ARPACK wrapper in SciPy
+            (scipy.sparse.linalg.svds), or “randomized” for the randomized
+            algorithm due to Halko (2009).
+
+        n_iter : int, optional (default 5)
+            Number of iterations for randomized SVD solver. Not used by ARPACK.
+            The default is larger than the default in randomized_svd to handle
+            sparse matrices that may have large slowly decaying spectrum.
+
+        random_state : int, RandomState instance or None, optional,
+                       default = None
+            If int, random_state is the seed used by the random number
+            generator; If RandomState instance, random_state is the random
+            number generator; If None, the random number generator is the
+            RandomState instance used by np.random.
+
+        tol : float, optional
+            Tolerance for ARPACK. 0 means machine precision. Ignored by
+            randomized SVD solver.
+
+        Attributes
+        ----------
+        columns_ : ndarray
+            The indices of the columns selected by the algorithm.
+        """
+        self.n_resamples = n_resamples
+        self.max_k = max_k
+        self.resample_frac = resample_frac
+        self.algorithm = algorithm
+        self.n_iter = n_iter
+        self.tol = tol
+        self.random_state = random_state
+
+    def fit(self, X, c=None, stratify=None):
```

Comment:
What does `c` do?

Reply:
It's the expected number of columns to select. I suppose it could go as a class attribute, but often you might want to change that quickly for the same dataset, so it's a hassle to have to reinitialize the object.